### PR TITLE
Add autofix for PYI009

### PR DESCRIPTION
**Pull request by qdegraaf**
Checked a recent autofix addition PR (https://github.com/charliermarsh/ruff/pull/4178/files#diff-6abdc1e4d11a89870a0a863b5132d8fa2f793889f56d1a1fa95fb0a0ce831486) for what I needed to change. 

If I understand `checker.patch` and autofix correctly, this should cover it. Happy to hear if I've missed anything.

Closes: https://github.com/charliermarsh/ruff/issues/4580

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pyi/rules/pass_statement_stub_body.rs [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 
-impl Violation for PassStatementStubBody {
+impl AlwaysAutofixableViolation for PassStatementStubBody {
     #[derive_message_formats]
     fn message(&self) -> String {
         format!("Empty body should contain `...`, not `pass`")
     }
+
+    fn autofix_title(&self) -> String {
+        format!("Replace `pass` with `...` in empty function body")
```
Comment (Skylion007) [inline crates/ruff/src/rules/flake8_pyi/rules/pass_statement_stub_body.rs:19]: Could be empty class or function body, right?

Comment (charliermarsh) [reply crates/ruff/src/rules/flake8_pyi/rules/pass_statement_stub_body.rs:19]: I think "Replace with `...`" is sufficient here

**Comment by charliermarsh** [comment]
Do you mind tweaking the suggestion and re-generating the test snapshots?

**Comment by charliermarsh** [comment]
(`cargo test`, then `cargo insta review`)

**Comment by qdegraaf** [comment]
> Do you mind tweaking the suggestion and re-generating the test snapshots?

I don't. Done!

**Pull request merged**